src/hooks/useApr.ts

```diff
@@ -0,0 +1,121 @@
+import { useStore } from 'src/store';
+import { computed, ref, watchEffect } from 'vue';
+import { useChainMetadata, useCurrentEra, useTvl } from '.';
+import { defaultAmountWithDecimals, reduceBalanceToDenom } from './helper/plasmUtils';
+import { useApi } from './useApi';
+
+// fixme: ideally get the value from API
+const STAKER_BLOCK_REWARD = 0.1;
+const DAPPS_BLOCK_REWARD = 0.4;
+
+const TS_FIRST_BLOCK = {
+  shiden: 1625570880, //  Ref: 2021-07-06 11:28:00 https://shiden.subscan.io/block/1
+  shibuya: 1630937640, // Ref: 2021-09-06 14:14:00 https://shibuya.subscan.io/block/1
+};
+
+export const useApr = () => {
+  const store = useStore();
+  const { api } = useApi();
+  const { decimal } = useChainMetadata();
+  const { tvlToken } = useTvl(api);
+  const { blockPerEra } = useCurrentEra();
+
+  const dapps = computed(() => store.getters['dapps/getAllDapps']);
+  const stakerApr = ref<number>(0);
+
+  const getAveBlocksPerMins = ({
+    chainName,
+    latestBlock,
+    timestampMillis,
+  }: {
+    chainName: string;
+    latestBlock: number;
+    timestampMillis: number;
+  }): number => {
+    if (chainName === 'shiden' || chainName === 'shibuya') {
+      const currentTs = Math.floor(timestampMillis / 1000);
+      const minsChainRunning = (currentTs - TS_FIRST_BLOCK[chainName]) / 60;
+      console.log('latestBlock:', latestBlock); //-> 709,400 blocks
+      console.log('minsChainRunning:', minsChainRunning); // -> 201,890.3 mins
+      const avgBlocksPerMin = latestBlock / minsChainRunning;
+      console.log('avgBlocksPerMin: 709,400 / 201,890.3 =', avgBlocksPerMin); // 3.51 blocks per mins
+      return avgBlocksPerMin;
+    }
+    return 0;
+  };
+
+  watchEffect(() => {
+    const apiRef = api && api.value;
+    const dappsRef = dapps.value;
+    const tvlTokenRef = tvlToken.value;
+    const decimalRef = decimal.value;
+    const blocksPerEraRef = Number(blockPerEra.value);
+
+    if (!apiRef || !dappsRef || !tvlTokenRef || !decimalRef || !blocksPerEraRef) {
+      return;
+    }
+
+    const getApr = async (): Promise<number> => {
+      console.log('getApr: Start ↓');
+      try {
+        const results = await Promise.all([
+          apiRef.query.balances.totalIssuance(),
+          apiRef.consts.blockReward.rewardAmount.toString(),
+          apiRef.runtimeVersion.specName.toString(),
+          apiRef.query.timestamp.now(),
+          apiRef.rpc.chain.getHeader(),
+        ]);
+
+        const rawTotalSupply = results[0];
+        const totalSupply = reduceBalanceToDenom(rawTotalSupply, decimalRef);
+        console.log('totalSupply:', totalSupply.toString()); // -> 71,001,295 SDN
+
+        const rawBlockRewards = results[1];
+        const blockRewards = Number(defaultAmountWithDecimals(rawBlockRewards, decimalRef));
+        console.log('blockRewards:', blockRewards.toString()); // -> 2.664 SDN
```

Comment:
the block reward of 2.664 SDN is divided to maintenance and dAppStaking in 50-50 ratio. Meaning, dAppsBlockReward = 1.332 SDN